internal/cmd/run.go

```diff
@@ -55,6 +55,54 @@ const (
 	waitForTracerProviderStopTimeout = 3 * time.Minute
 )
 
+// serve starts the REST API server and ensures it is properly shut down when
+// the srvCtx is done. It returns a function that can be waited on to ensure
+// the server has fully stopped.
+//
+// nolint:nestif
+func (c *cmdRun) serve(cmd *cobra.Command, srv *http.Server, shutdown func()) func() {
```

Comment:
This doesn't seem necessary to pass the full `Command` object to the function, what about just having an `addressFlagChanged bool` param? (name could probably be improved :sweat_smile:)